Rewrite the "Adding Custom Hitsounds" guide (#3127)
ppy/osu-wiki: wiki/Guides/Adding_Custom_Hitsounds/en.md

Note that you do not need to add a custom hitsound for every sound in a sampleset. You will notice in the first example image that no instance of `soft-slidertick.wav` is present, and in this case, osu! will continue to use the default sound for all successful regular slider tick hits when the Soft sampleset is in use.

## Working with multiple hitsounds in the same sampleset

Sometimes, a song may have multiple sections with very different musical styles, and one group of hitsounds just doesn't fit both sections. In these cases, it is often helpful to utilise a different hitsound (or group of hitsounds) entirely. This can be achieved by adding a number to the end of the hitsound file name as follows:

`<sampleset>-<sound><#>.wav`

where `<#>` can be any number of your choice. The osu! editor natively supports values between 2 and 100, but greater values can be achieved via editing of the `.osu` file if required. Note that the first group of hitsounds does not need to be labelled with the number "1" even if multiple hitsound groups are used, so `soft-hitclap1.wav` won't work, and `soft-hitclap.wav` should continue to be used instead.

> In order to ensure that different-numbered hitsounds or hitsound groups play correctly, you'll need to add an inherited timing point (green line) and change the sampleset from "Custom 1" to the option immediately below it, as shown in the example below. Here you can enter the number of the hitsound group defined earlier which you wish to use.

Reviewer: ```suggestion
In order to ensure that different-numbered hitsounds or hitsound groups play correctly, you'll need to add an inherited timing point (green line) and change the sampleset from "Custom 1" to the option immediately below it, as shown in the example below. Here you can enter the number of the hitsound group which you want to use.
```

![](img/CustomHSAddition_03.png "Changing to a second custom hitsound group.")

> ### Example 2

Reviewer: ```suggestion
### Example
```

When the sampleset of an inherited timing point is set to `S:C2` as in the example image above, all default hitsounds and hitsound additions will be replaced by the appropriately named custom hitsounds, such as `soft-hitclap2.wav` where present. These will continue to take effect until an inherited timing point with a different sampleset is encountered—in this example at `02:00:723 -`, when the sampleset switches back to `S:C1`.

## External sources

> - ["how to add custom hitsound?" forum thread by neonat](https://osu.ppy.sh/community/forums/posts/3215699)

Reviewer: ```suggestion
- ["how to add custom hitsound?"](https://osu.ppy.sh/community/forums/posts/3215699)  forum thread by [neonat](https://osu.ppy.sh/users/1561995)
```
might be better

Once you have your desired sound files, they should be moved into the song folder of the beatmap you would like to use them in. If you don't know where your song folder is located, follow these directions:
1. Open osu!
2. Select the *Edit* option
3. Navigate to *your beatmap* and open it
4. Click *File* (the left-most option in an off-white coloured bar at the top)
5. Click *Open Song Folder*
6. *Paste your files here*

Once the sound file(s) have been placed inside the beatmap's Song Folder, they must be named appropriately in order for osu! to recongnize them as *hitsounds*.

Three basic categories (or *samplesets*) of hitsound exist in osu!: Normal (N), Soft (S), and Drum (D). Each sampleset can be further subdivided into different sounds. The most common of these are the "hitnormal", "hitclap", "hitwhistle" and "hitfinish" hitsounds, although some more specialized sounds exist, such as those that play during a slider ("sliderslide", "slidertick") or spinner ("spinnerspin").

Hitsound files are named to reflect these two properties of sampleset and the type of sound as follows:

> `<sampleset><sound>.wav`

Reviewer 2: Probably needs a `-` here, so

`<sampleset>-<sound>.wav`

> where <sampleset> is either "normal", "soft" or "drum", and <sound> is one of the examples mentioned above, e.g. "hitclap".

Reviewer 2: Wrap `<sampleset>` and `<sound>` with grave keys (`) for them to appear as in-line code.

> ![An example of custom hitsound naming convention inside a beatmap's Song Folder](ACH_01.png "Tutorial Image 1")

Reviewer 2: We should get an updated image which has file extensions on them.

(also it's kinda outdated because mp3 files are all called audio.mp3 by default)

Author: This has now been replaced.

### Example 1

> In the example shown above, the first sound listed is a `normal-hitclap.wav`, and will replace the default hitsound that plays when a note with the Normal sampleset, hitclap addition is successfully hit. Note that this will only play on the selected sampleset—if your beatmap uses other samplesets, they will require additional hitsound files (even if you intent to use the exact same sound sample, such as is done using `soft-hitclap.wav` here.

Reviewer 2: `intent` -> `intend`

Reviewer 2: Also forgot to close the bracket.

![Tutorial Image 2](ACH_02.png "Tutorial Image 2")

In order to have osu! play your custom hitsounds, make sure to check the second option "Custom 1" in the image shown above. Note that you do not need to add a custom hitsound for every sound in a sampleset. You will notice in the first example image that no instance of `normal-hitnormal.wav` is present, and in this case, osu! will continue to use the default sound for all successful regular hits (without hitsounds additions) when the Normal sampleset is in use.

## Working With Multiple Hitsounds In The Same Sampleset

> Sometimes, a song might benefit from more variety in hitsounds, such as when there are many different sounds of a single instrument (e.g. a drum fill or a guitar solo) present. Alternatively, a song may have multiple sections with very different musical styles, and one group of hitsounds just doesn't fit both sections.

Reviewer 2: Would probably simplify this first part of the first sentence

> Sometimes, a song may require the use of many hitsounds, such as...

In these cases, it is often helpful to utilise a different hitsound (or group of hitsounds) entirely. This can be achieved by adding a number to the end of the hitsound file name as follows:

`<sampleset><sound><#>.wav`

> where <#> can be any number between 2 and 100. Note that the first group of hitsounds does not need to be labelled with the number "1" even if multiple hitsound groups are used, so `normal-hitclap1.wav` won't work, and `normal-hitclap.wav` should continue to be used instead.

Reviewer 2: Would inline code the `<#>` too

Reviewer 2: Also I *think* you can have any integer number above 1 (you just have to edit the .osu to be able to set timing lines above 99), will need to test this

Reviewer 2: Last sentence, would drop the `continue to`, so

> and `normal-hitclap.wav` should be used instead.

> In order to ensure that different-numbered hitsounds or hitsound groups play correctly, you'll need to add an inhereited timing point (green line) and change the sampleset from "Custom 1" to the option immediately below it, where you can enter the number of the hitsound group you wish to use.

Reviewer 2: `inhereited ` -> `inherited`

### Example 2

> 

Reviewer 2: We should get an image for this example too.

Author: Agreed as stated in the PR, I'll get this done tonight.

Author: New images added, let me know if the quality is lacking and I'll do something on a higher res monitor at work tomorrow,

> When the sampleset of an inhereited timing point is set to "N:C2" all default hitsounds and hitsound additions will be replaced by appropriately named custom hitsounds, such as `normal-hitclap2.wav` where present, until an inherited timing point with a different sampleset is encountered.

Reviewer 2: Would inline code `N:C2`

Reviewer 2: will be replaced by appropriately named custom hitsounds -> will be replaced by **the** appropriately named custom ~~hitsounds~~ hitsound

Reviewer 2: Would cut out the `such as `normal-hitclap2.wav` where present` for now and add it to a paragraph where we can describe how this works better with some images.

Author: Fixed the first two, will keep the last where it is for now and reword it as necessary with the new image.

From the thread: *[how to add custom hitsound? by: neonat](https://osu.ppy.sh/community/forums/posts/3215699)* (and modified)

## Custom Hitsound Files

In order to use custom hitsound files in osu!, you'll first need some custom hitsounds! The [Custom Hitsound Library](../Custom_Hitsound_Library/) is a great resource for existing sound samples, including cymbals, drums, bells, whistles and more! Alternatively, if you can't find what you're looking for, you can make your own samples!

> Hitsounds should typically be in the `.wav` or `.ogg` format, as these have no playback delay and loop correctly. `.mp3` formatted sound files have slight delays and may not accurately match up to the sounds in the music you are trying to hitsound. However, they but may still be useable for certain special hitsounds such as applause or ambient noises, where the size of `.wav` or `.ogg` files may be prohibitively large.

Reviewer 3: wording is a bit messed up: `However, they but may still...`
should delete the but

Reviewer 3: adding a comma to: `certain, special hitsounds` would be more grammatically correct

Once you have your desired sound files, they should be moved into the song folder of the beatmap you would like to use them in. If you don't know where your song folder is located, follow these directions:
1. Open osu!

> 2. Select the *Edit* option

Reviewer 4: This method of reaching the song folder not work on osx builds afaik, a different process is needed:
1. Right click on the Osu! application icon and click "Show Package Contents"
2. drive_c -> osu! -> Songs -> find the song folder 
(Sorting by "Last Modified" is a good idea)

Author: Was not aware of this, clarified with some MacOS users and have added additional steps here as you suggested.

3. Navigate to *your beatmap* and open it
4. Click *File* (the left-most option in an off-white coloured bar at the top)
5. Click *Open Song Folder*
6. *Paste your files here*

Once the sound files have been placed inside the beatmap's Song Folder, they must be named appropriately in order for osu! to recongnise them as *hitsounds*.

> Three basic categories of hitsound, refered to as *samplesets*, exist in osu!: Normal (N), Soft (S), and Drum (D). Each sampleset can be further subdivided into different sounds. The most common of these are the "hitnormal", "hitclap", "hitwhistle" and "hitfinish" hitsounds, although some more specialized sounds exist, such as those that play during a slider ("sliderslide", "slidertick") or spinner ("spinnerspin").

Reviewer 3: `categories of hitsounds, referred to` needs an s and another r in refered

Reviewer 3: This sentence is really long. Consider splitting it up like so: `The most common of these are the "hitnormal", "hitclap", "hitwhistle" and "hitfinish" hitsounds. More specialized sounds also exist, such as those that play during a slider ("sliderslide", "slidertick") or a spinner ("spinnerspin").`

Author: Agreed, split it up
